spec/javascripts/student_profile/restricted_notes_page_container_spec.js

```diff
@@ -0,0 +1,59 @@
+//= require ./fixtures
+
+describe('RestrictedNotesPageContainer', function() {
+  var dom = window.shared.ReactHelpers.dom;
+  var createEl = window.shared.ReactHelpers.createEl;
+  var merge = window.shared.ReactHelpers.merge;
+
+  var SpecSugar = window.shared.SpecSugar;
+  var Fixtures = window.shared.Fixtures;
+  var RestrictedNotesPageContainer = window.shared.RestrictedNotesPageContainer;
+
+  var helpers = {
+    renderInto: function(el, props) {
+      var mergedProps = merge(props || {}, {
+        nowMomentFn: moment.utc,
+        serializedData: Fixtures.studentProfile,
+      });
+      return ReactDOM.render(createEl(RestrictedNotesPageContainer, mergedProps), el);
+    },
+
+    createMockApi: function(){
+      var mockApi = jasmine.createSpyObj('api', ['saveNotes']);
+      // We need to mock a promise object -- this is a poor man's solution.
```

Comment:
This is a start of a good solution, and these are great tests! :)

Here you have to know the order of calls (`.done` chained with `.fail`) since it's just a stub object.  You can use real Promises here with fake values though, something like:

```
var mockResponse = {id: 781, text: 'hi', is_restricted: true, event_note_type_id: 301, student_id: 23};
mockApi.saveNotes.and.returnValue($.Deferred().resolve(mockResponse);
```

That way code like https://github.com/erose/studentinsights/blob/simplify-restricted-notes/app/assets/javascripts/restricted_notes/restricted_notes_page_container.js#L76 will also get exercised all the way through with real values.